File: 0301_0400/0313/0313.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.readline())
    m = [0 for i in range(101)]
    k = 0
    s = [int(x) for x in f.readline().split()]
    for i in range(1, n + 1):
        if m[s[i - 1]] and i - m[s[i - 1]] > k:
            k = i - m [s[i - 1]]
        m[s[i - 1]] = i
    q.write(str(k))

File: 0301_0400/0313/0313.c
#include <stdio.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, k = 0, m[101] = {0}, a;

	fscanf(f, "%d", &n);
	for (int i = 1; i <= n; ++i)
	{
		fscanf(f, "%d", &a);
		if (m[a] && i - m[a] > k)
			k = i - m[a];
		m[a] = i;
	}
	fprintf(q, "%d", k);
	return 0;
}
